service/assessment/assessment.go

```diff
@@ -106,8 +109,8 @@ type Service struct {
 
 	authorizer api.Authorizer
 
-	// grpcOpts contains additional gRPC options that will be appended to all gRPC dial calls. Useful for testing.
-	grpcOpts []grpc.DialOption
+	// grpcTargets contains the target and DialOptions
+	grpcTargets map[string]grpcTarget
```

Comment:
Having this as a map introduces another layer of errors that needs to be caught (missing keys), etc. Personally, I would recommend just replacing the original `orchestratorAddress` and `evidenceStoreAddress` with the new internal struct.

Reply:
Ok, I thought so we could add easily more services.

Reply:
A good idea in theory but then we would need to also remove `orchestratorClient` and streams etc. so I think we can keep it as two separate "addresses" for now. Let's not break too many things at once ;)